spec/views/user_index_spec.rb

```diff
@@ -0,0 +1,29 @@
+require 'rails_helper'
+
+RSpec.describe 'User', type: :system do
+  describe 'index page' do
+    it 'shows the username of all users' do
+      visit users_path
+      expect(page).to have_content('Wassim Chakib')
+      expect(page).to have_content('Mert Kaya')
+    end
+
+    it 'shows the pictures of all other users' do
+      visit users_path
+      expect(page).to have_selector('img')
+    end
+
+    it 'shows the number of posts of all other users' do
+      visit users_path
+      expect(page).to have_content('Number of posts: 3')
+      expect(page).to have_content('Number of posts: 1')
```

Comment:
- [ ] It would be better if we address all the failing test cases. *Hint:* It would be better if we create a `before :each`. Like for example
```
before :each do
    User.create(email: 'user@example.com', password: 'password')
end

```
Then we can now compare the attributes to the expected outcome. In our case, what are we comparing to?
![Screenshot from 2023-03-08 08-35-42](https://user-images.githubusercontent.com/46388113/223663721-5912f06c-4912-495f-a070-58491e944eb2.png)

Reply:
Thank you for suggestions. But I think before each is not that good practice since it is saving the data to database, Instead you can use seeds.rb by using the command `rails db:seed RAILS_ENV=test` to populate the test database with dummy data and you can easily make the test. Before each is not consistent. And if you configure database with the dummy data correctly, you should get this response below: 
![image](https://user-images.githubusercontent.com/50754458/223668664-e7333daa-7dc2-4afa-a654-e0218daf117f.png)

Reply:
Another thing, I want to talk about is the screenshots of n+1 are in the pull request already. Please try to read Pull request carefully and open the screenshots above here....

![image](https://user-images.githubusercontent.com/50754458/223669024-b073a226-f104-490d-ad27-ff1577eba7f8.png)

Reply:
oh yeah I surely missed that. Sorry about that

Reply:
> Thank you for suggestions. But I think before each is not that good practice since it is saving the data to database, Instead you can use seeds.rb by using the command `rails db:seed RAILS_ENV=test` to populate the test database with dummy data and you can easily make the test. Before each is not consistent. And if you configure database with the dummy data correctly, you should get this response below: ![image](https://user-images.githubusercontent.com/50754458/223668664-e7333daa-7dc2-4afa-a654-e0218daf117f.png)

![Screenshot from 2023-03-08 09-12-23](https://user-images.githubusercontent.com/46388113/223672126-ccf15aa4-b65f-4549-8090-f945e09fe1e0.png)
Verified